odoo/orm/domains.py

```diff
@@ -364,6 +367,17 @@ def validate(self, model: BaseModel) -> None:
         # just execute the optimization code that goes through all the fields
         self.optimize(model, full=True)
 
+    def filter_records(self, records: M) -> M:
```

Comment:
When designing an API, I always imagine what calls will look like.  And this doesn't look good to me:
```py
records = domain.filter_records(records)  # "records" is redundant
predicate = domain._filter_records(records)  # this does not filter records
```
I suggest this instead:
```py
records = domain.filter(records)  # you can only filter records with a domain
predicate = domain._as_predicate(records)  # convert the domain to a predicate function
```